plugins/cairo-lang-macro-stable/src/lib.rs

```diff
@@ -15,21 +18,52 @@ pub enum StableAuxData {
     Some(*mut c_char),
 }
 
+/// Diagnostic returned by the procedural macro.
+///
+/// This struct implements FFI-safe stable ABI.
+#[repr(C)]
+#[derive(Debug)]
+pub struct StableDiagnostic {
+    pub message: *mut c_char,
+    pub severity: StableSeverity,
+}
+
+/// The severity of a diagnostic.
+///
+/// This struct implements FFI-safe stable ABI.
+#[repr(C)]
+#[derive(Debug)]
+pub enum StableSeverity {
+    Error,
+    Warning,
+}
+
```

Comment:
I think this should be just

```rust
type StableSeverity = NonZero<u8>;
```

By using enums you missed two erroneous cases:
1. Scarb introduced a new severity level, but the plugin is using old API
2. Plugin is using new API with new severity level, but it's being loaded into old Scarb.